ophyd/v2/core.py

```diff
@@ -484,16 +465,18 @@ async def get_reading(self) -> Reading:
     async def get_value(self) -> T:
         return self._value
 
-    def monitor_reading_value(self, callback: ReadingValueCallback[T]) -> Monitor:
-        callback(self._reading, self._value)
-        return SimMonitor(callback, self.listeners)
+    def set_callback(self, callback: Optional[ReadingValueCallback[T]]) -> None:
+        if callback:
+            callback(self._reading, self._value)
+
+        self.callback = callback
```

Comment:
E.g.
```suggestion
    def set_callback(self, callback: Optional[ReadingValueCallback[T]]) -> None:
        if callback:
            assert not self.callback, "Cannot set a callback when one is already set"
            callback(self._reading, self._value)
        self.callback = callback
```

Reply:
should I also apply similar logic for the aioca/p4p backends?